ops/jenkins/bfd-build-apps/Jenkinsfile

```diff
@@ -0,0 +1,175 @@
+#!/usr/bin/env groovy
+
+/**
+ * <p>
+ * This script will be run by Jenkins when building apps and app AMIs.
+ * </p>
+ */
+
+// These variables are accessible throughout this file (except inside methods and classes).
+def verboseMaven = params.verbose_mvn_logging
+def appBuildResults
+def amiIds
+def scriptForApps
+def scriptForDeploys
+
+// send notifications to slack, email, etc
+def sendNotifications(String buildStatus = '', String stageName = '', String gitCommitId = '', String gitRepoUrl = '') {
+    // buildStatus of NULL means success
+    if (!buildStatus) {
+        buildStatus = 'SUCCESS'
+    }
+
+    // build colors
+    def colorMap = [:]
+    colorMap['STARTED']  = '#0000FF'
+    colorMap['SUCCESS']  = '#00FF00'
+    colorMap['ABORTED']  = '#6A0DAD'
+    colorMap['UNSTABLE'] = '#FFFF00'
+    colorMap['FAILED']   = '#FF0000'
+    def buildColor = colorMap[buildStatus]
+    buildColor = buildColor ?: '#FF0000' // default to red
+
+    def branchUrl = "https://github.com/CMSgov/beneficiary-fhir-data/tree/${params.git_branch}"
+
+    def msg = ''
+    switch (buildStatus) {
+        case 'UNSTABLE':
+        case 'SUCCESS':
+            msg = 'COMPLETED SUCCESSFULLY: BUILT APPS AND AMIS'
+            break
+        case 'FAILED':
+        case 'FAILURE':
+            msg = "FAILED ON ${stageName.toUpperCase()} STAGE ON BRANCH <${branchUrl}|${params.git_branch.toUpperCase()}>"
+            break
+        case 'STARTED':
+            msg = 'HAS STARTED'
+            break
+        case 'ABORTED':
+            msg = 'WAS ABORTED'
+            break
+        default:
+            msg = "${buildStatus.toUpperCase()}"
+            break
+    }
+
+    // build slack message
+    def slackMsg = ''
+    if (buildStatus == 'UNSTABLE') {
+        slackMsg = "UNSTABLE APP BUILD <${env.BUILD_URL}|#${env.BUILD_NUMBER}> ON BRANCH <${branchUrl}|${params.git_branch.toUpperCase()}> ${msg} \n"
+    } else {
+        slackMsg = "APP BUILD <${env.BUILD_URL}|#${env.BUILD_NUMBER}> ON BRANCH <${branchUrl}|${params.git_branch.toUpperCase()}> ${msg} \n"
+    }
+    slackMsg += "\tJob '${env.JOB_NAME}' \n"
+
+    // send Slack messages
+    slackSend(color: buildColor, message: slackMsg)
+}
+
+pipeline {
+    agent {
+        kubernetes {
+            defaultContainer 'bfd-cbc-build'
+            yaml """
+apiVersion: v1
+kind: Pod
+spec:
+  serviceAccount: bfd
+  containers:
+  - name: bfd-cbc-build
+    image: 'public.ecr.aws/c2o1d8s9/bfd-cbc-build:jdk17-mvn3-tfenv3-latest'
+    command:
+    - cat
+    tty: true
+    imagePullPolicy: Always
+    resources:
+      requests:
+        memory: '16384Mi'
+        cpu: '8000m'
+      limits:
+        memory: '16384Mi'
+        cpu: '8000m'
+"""
+        }
+    }
+
+    parameters {
+        string(name: 'git_branch',
+        defaultValue: '',
+        description: 'To build, specify a branch within BFD.')
```

Comment:
This `git_branch` parameter needs to be more functional in the build. Overall, we'd like to try something a _little_ different from what we have been doing in the current multibranch/multistage pipeline(s) where the git reference is assumed (see the unqualified `checkout scm` step below). So instead of taking the assumed git context from the job, we want this job to enable us to build a specific git reference.

Reply:
Took your and mitch's advice on how to implement this